### PR TITLE
Proof of concept - chatting between staff & adopters

### PR DESCRIPTION
**Pull request by edwinthinks**
### Describe your change in plain English.

Wanted to see what I whip up real quick based on our conversation with Danielle about maybe wanting to chat with adopters and staff. Maybe in the future include foster as well.

The main point of this is to see if we want to pursue more time to refining this feature.

https://www.loom.com/share/5ee3b313a00641cdb53412d5405ea9a7

### Link to the issue

N/A

### CODE REVIEW

File: app/controllers/adopter_applications/chats_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AdopterApplications::ChatsController < ApplicationController
+  def show
+    @match = AdopterApplication.find(params[:adopter_application_id])
```
Comment (Thrillberg) [inline app/controllers/adopter_applications/chats_controller.rb:3]: Why are we using `AdopterApplication` instead of `Adoption` here? Seems intuitive to use `Adoption`.

File: app/javascript/controllers/chat_message_controller.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Controller } from "@hotwired/stimulus"
+
+/**
+ * Add mutation to move the scroll to the bottom of the chat when
+ * it is updated.
+ */
+
+export default class extends Controller {
+  connect() {
+    this.element.scrollIntoView()
```
Comment (Thrillberg) [inline app/javascript/controllers/chat_message_controller.js:10]: This is rad!

File: test/models/message_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class MessageTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
```
Comment (Thrillberg) [inline test/models/message_test.rb:5]: yeah but is the truth really true 🤔

**Review by Thrillberg — APPROVED** [review]
Looks great! Where in the logic does it determine who can see the messages?

File: app/controllers/adopter_applications/chats_controller.rb (hunk 1 above)
Comment (edwinthinks) [reply app/controllers/adopter_applications/chats_controller.rb:3]: Good question, I think `Adoption` makes more sense as you said. I don't have any strong ties to the choice to use AdopterApplication. Whatever we use, it just needs to be related to the Adopter & Staff.

File: app/javascript/controllers/chat_message_controller.js (hunk 2 above)
Comment (edwinthinks) [reply app/javascript/controllers/chat_message_controller.js:10]: Jeje it is good, but actually has some issues -- which it scrolls down the entire page rather than just the internal element. I'd probably use something like -- https://github.com/stimulus-use/stimulus-use/blob/main/docs/use-mutation.md to control both the scrolling down and moving the element to the left or right depending on who sent the message.

File: test/models/message_test.rb (hunk 3 above)
Comment (edwinthinks) [reply test/models/message_test.rb:5]: Who knows, in real life this assertion is actually flaky lol

**Comment by edwinthinks** [comment]
> Looks great! Where in the logic does it determine who can see the messages?

There is no logic yet for this since its a proof of concept. However, the way to do it later would be to put authorization at the controller level to prevent access to the page. Within the page adding `turbo_stream_from` handles broadcasting/subscribing.

**Comment by edwinthinks** [comment]
@Thrillberg thanks for the review! I'am not actually planning on merging this until we have some sort of UI figured out. Then we can re-open and maybe let another use this as a reference to implement. Great for any gooders who want to learn hotwire in a live setting!
